controllers/scan.js

```diff
@@ -21,27 +23,33 @@ async function post (req, res) {
   });
 
   if (req.body.featuredBreach) {
-    let featuredBreach = req.body.featuredBreach;
-    if(foundBreaches.filter(breach => breach.Name === featuredBreach).length > 0) {
-      featuredBreach = foundBreaches.filter(breach => breach.Name === featuredBreach);
+    featuredBreach = req.app.locals.breaches.filter(breach => breach.Name.toLowerCase() === req.body.featuredBreach.toLowerCase())[0];
```

Comment:
Same as above? Should we use `.find()`?

https://github.com/mozilla/blurts-server/blob/5cb8a1cc8467dd45b0201c05b2729184fcfad2cf/controllers/hibp.js#L12

Reply:
I mean... if we're using this pattern in at least 3-4 places, we should probably consider turning it into a helper somewhere. Like `req.app.locals.getBreach(req.body.featuredBreach)`

Reply:
Actually, this is what I was thinking, possibly in HIBP.js:

```js
app.locals.getBreach = (name) => breaches.find(breach => breach.Name.toLowerCase() === name.toLowerCase());
```

Then to consume it, it'd just be the `req.app.locals.getBreach(req.body.featuredBreach);` (or whatever).
But now that I type that out, I'm wondering if a lookup is the best solution, or if `app.locals.breaches` should be a `Map()` so we could just do `app.locals.breaches.has(req.body.featuredBreach)` or `app.locals.breaches.get(reqBreachName)` instead of having to loop.

```sh
$ git grep "app.locals.breaches."

controllers/hibp.js:  const breach = req.app.locals.breaches.find(breach => breach.Name.toLowerCase() === reqBreachName);
controllers/home.js:    featuredBreach = req.app.locals.breaches.filter(breach => breach.Name.toLowerCase() === reqBreachName)[0];
controllers/scan.js:  let foundBreaches = await HIBP.getBreachesForEmail(emailHash, req.app.locals.breaches);
controllers/scan.js:    featuredBreach = req.app.locals.breaches.find(breach => breach.Name.toLowerCase() === req.body.featuredBreach.toLowerCase());
hibp.js:      app.locals.breaches = breaches;
```

Or maybe that's just getting silly.